### PR TITLE
Avoid re-copying include headers

### PR DESCRIPTION
**Pull request by abrown**
This adds a `.stamp` file to the build directory that tracks when all headers have been copied over to the sysroot `include` directory. Previously, the `include_dirs` target was phony, which `make` will run every time, regardless of whether it is needed. This was part of the problem with all of libc being rebuilt at each `make` invocation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -836,7 +845,11 @@ ifeq ($(WASI_SNAPSHOT), p2)
 		> "$(SYSROOT_INC)/__wasi_snapshot.h"
 endif
```
Comment (abrown) [inline Makefile:846]: @sunfishcode, I actually have a Bash script that we could also use here to move out all of these commands out of the Makefile along with the list of `MUSL_OMIT_HEADERS`. Should I bring that in to this PR?

Comment (sunfishcode) [reply Makefile:846]: That sounds like a good change; would you mind submitting a separate PR for that?

**Review by sunfishcode — APPROVED** [review]
This looks good to me; it just needs a merge conflict resolution.

File: Makefile (hunk 1 above)
Comment (abrown) [reply Makefile:846]: #552

**Pull request merged**